extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { NotebookInfo } from '../interfaces';
+import { isString } from 'util';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+export interface INotebookService {
+	launchNotebook(notebook: string | NotebookInfo): void;
+}
+
+export class NotebookService implements INotebookService {
```

Comment:
Does this (and the other services) really need to be actual classes? You're only going to be creating one of them so it doesn't really seem necessary vs just having the functions defined by themselves.

Reply:
there are different ways of implementing this, I prefer this way so that i can easily mock it, both for testing and to make parallel development easier.